dll/win32/msgina/shutdown.c

```diff
@@ -1029,9 +1030,18 @@ ShutdownDialogProc(
             ShutdownOnInit(hDlg, pContext);
 
             /* Draw the logo bitmap */
-            pContext->hBitmap =
-                LoadImageW(pContext->pgContext->hDllInstance, MAKEINTRESOURCEW(IDI_ROSLOGO), IMAGE_BITMAP, 0, 0, LR_DEFAULTCOLOR);
-            return TRUE;
+            if (IsWindowsServer())
+            {
+                pContext->hBitmap =
+                    LoadImageW(pContext->pgContext->hDllInstance, MAKEINTRESOURCEW(IDI_ROSLOGO_SERVER), IMAGE_BITMAP, 0, 0, LR_DEFAULTCOLOR);
+                return TRUE;
+            }
+            else
+            {
+                pContext->hBitmap =
+                    LoadImageW(pContext->pgContext->hDllInstance, MAKEINTRESOURCEW(IDI_ROSLOGO_WORKSTATION), IMAGE_BITMAP, 0, 0, LR_DEFAULTCOLOR);
+                return TRUE;
+            }
```

Comment:
```suggestion
            if (IsWindowsServer())
            {
                pContext->hBitmap = LoadImageW(pContext->pgContext->hDllInstance,
                                               MAKEINTRESOURCEW(IDI_ROSLOGO_SERVER), IMAGE_BITMAP,
                                               0, 0, LR_DEFAULTCOLOR);
            }
            else
            {
                pContext->hBitmap = LoadImageW(pContext->pgContext->hDllInstance,
                                               MAKEINTRESOURCEW(IDI_ROSLOGO_WORKSTATION), IMAGE_BITMAP,
                                               0, 0, LR_DEFAULTCOLOR);
            }
            return TRUE;
```

Reply:
Ideally it would be abstract this code somewhere, since we already have it in shell32 right? :)

Reply:
Besides the fact that msgina should be able to run in the absence of shell32...
On Vista+ there's a specific resource-only branding dll just for this kind of thing.